packages/config/src/env/main.js

```diff
@@ -96,6 +97,23 @@ const getGeneralEnv = async function ({
   })
 }
 
+const getDeployUrls = function ({
+  siteInfo: { name = DEFAULT_SITE_NAME, ssl_url: sslUrl, build_settings: { repo_url: REPOSITORY_URL } = {} },
+  branch,
+  deployId,
+}) {
+  return {
+    URL: sslUrl,
+    REPOSITORY_URL,
+    DEPLOY_PRIME_URL: `https://${branch}--${name}${NETLIFY_DEFAULT_DOMAIN}`,
+    DEPLOY_URL: `https://${deployId}--${name}${NETLIFY_DEFAULT_DOMAIN}`,
+  }
+}
+
+const NETLIFY_DEFAULT_DOMAIN = '.netlify.app'
+// `site.name` is `undefined` when there is no token or siteId
+const DEFAULT_SITE_NAME = 'site-name'
```

Comment:
In local builds, this default value for the site name is used by the `DEPLOY_URL` and `DEPLOY_PRIME_URL` environment variables, but not by the `SITE_NAME` environment variable. This is to be consistent with `SITE_ID` which is also unavailable when the authentication token or `siteId` has not been provided. Please note this situation is an edge case, since it cannot happen in neither production nor CLI builds, only when `@netlify/build` is called programmatically (which at the moment is mostly only used in `@netlify/build` automated tests).